src/authorizationProvider.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from "react";
+import authService from "./services/authService";
+
+const refreshToken = 'RefreshToken';
+const accessToken = 'AccessToken';
+const AuthorizationContext = React.createContext();
+class AuthorizationProvider extends Component {
+    state = {
+        loggedIn: null,
+        userFirstName: null,
+        userLastName: null,
+        userRole: null
+    }
+
+    async componentDidMount() {
+        if (localStorage.getItem(accessToken) !== 'null' &&
+            localStorage.getItem(refreshToken) !== 'null' &&
+            localStorage.getItem(accessToken) !== null &&
+            localStorage.getItem(refreshToken) !== null) {
+            const response = await authService.refresh(localStorage.getItem(accessToken), localStorage.getItem(refreshToken));
+
+            if (response.data.succeeded === true) {
+                let jwtData = response.data.accessToken.split('.')[1];
+                let stringJwtJsonData = window.atob(jwtData);
+                let decodedJwt = JSON.parse(stringJwtJsonData);
+                let role = decodedJwt['http://schemas.microsoft.com/ws/2008/06/identity/claims/role'];
+                localStorage.setItem(accessToken, response.data.accessToken);
+                localStorage.setItem(refreshToken, response.data.refreshToken);
+
+                this.setState({
+                    loggedIn: true,
+                    userFirstName: response.data.userFirstName,
+                    userLastName: response.data.userLastName,
+                    userRole: role
+                });
+            }
+        }
+    }
+
+    render() {
+        return (
+            <AuthorizationContext.Provider value={{
```

Comment:
tu plati rovnaka poznamka s `value` ako v tomto komentari, a teda najlepsie je do `value` posielat len `this.state`

Reply:
done